scrumcube/pj-movie
# Forgot and change password

# Forgot and Change password
- Admin
    1. Generate controller: controllers/admin/
        - [x] passwords_controller.rb
           `- new action`
           `- edit action`
           `- customize def after_resetting_password_path_for`
    2. Update file: db/migrate/
        - [x] 20191211013853_devise_create_admins.rb
           `- add field t.integer  :sign_in_count, default: 0, null: false`
          ` - t.datetime :current_sign_in_at`
          `- t.datetime :last_sign_in_at`
          `- t.string   :current_sign_in_ip`
         ` - t.string   :last_sign_in_ip`
   3. Add new file in: views/admin/passwords/
        - [x] new.html.erb
           `- prepare form design html`
        - [x] edit.html.erb
           `- prepare form design html`
   4. Add new file in: spec/request/admin/
        - [x] passwords_controller_spec.rb
           `- test reset`
           `- test should get confirm email form`

- Users
    1. Generate controller: controllers/users/
        - [x] passwords_controller.rb
           `- new action`
           `- edit action`
    2. Update file: db/migrate/
        - [x] 20191205073658_devise_create_users.rb
           `- add field t.integer  :sign_in_count, default: 0, null: false`
          ` - t.datetime :current_sign_in_at`
          `- t.datetime :last_sign_in_at`
          `- t.string   :current_sign_in_ip`
         ` - t.string   :last_sign_in_ip`
   3. Add new file in: views/users/passwords/
        - [x] new.html.erb
           `- prepare form design html`
        - [x] edit.html.erb
           `- prepare form design html`
   4. Add new file in: spec/request/user/
        - [x] passwords_controller_spec.rb
           `- test reset`
           `- test should get confirm email form`
